src/hotspot/share/cds/filemap.cpp

```diff
@@ -1565,11 +1567,36 @@ static size_t write_bitmap(const CHeapBitMap* map, char* output, size_t offset)
   return offset + size_in_bytes;
 }
 
+// The start of the archived heap has many primitive arrays (String
+// bodies) that are not marked by the oop/ptr maps. So we must have
+// lots of leading zeros.
+size_t FileMapInfo::remove_bitmap_leading_zeros(CHeapBitMap* map) {
+  size_t old_zeros = map->find_first_set_bit(0);
+  size_t old_size = map->size_in_bytes();
+
+  // Slice and resize bitmap
+  map->truncate(old_zeros, map->size());
+
+  // We want to keep track of how many zeros were removed
+  size_t new_zeros = map->find_first_set_bit(0);
```

Comment:
I think this statement could be enclosed with `DEBUG_ONLY` since the `new_zeros` is only used in the `assert `statement following it.